pkg/reconciler/route/route.go

```diff
@@ -302,13 +313,32 @@ func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 		Hostname: resourcenames.K8sServiceFullname(r),
 	}
 
+	tls := []netv1alpha1.ClusterIngressTLS{}
+	if config.FromContext(ctx).Network.AutoTLS {
+		domains := tr.GetDomains(r.Status.URL.Host, traffic.Targets)
+		desiredCert := resources.MakeCertificate(r, domains)
+		cert, err := c.reconcileCertificate(ctx, r, desiredCert)
+		if err != nil {
+			r.Status.MarkCertificateProvisionFailed(desiredCert.Name)
+			return err
+		}
+
+		if cert.Status.IsReady() {
+			r.Status.MarkCertificateReady(cert.Name)
```

Comment:
I think that here we should probably upgrade the scheme of the URLs that are in-scope for auto-TLS.

Reply:
Not that at some point (with "Kevins") we will want to support a mix of cluster-local and normal URLs, so we should only auto-TLS and change the scheme on normal URLs.

Reply:
Currently I marked all URLs as HTTPS. Added a TODO here to support a mix of cluster-local and normal URLs when we are able to configure visibility per target.